File: apps/delivery_partner_app/lib/models/auth_response.dart
class AuthResponse {
  AuthResponse({required this.token, required this.role});

  final String token;
  final String role;

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    final data = json['data'] as Map<String, dynamic>;
    return AuthResponse(
      token: data['token'] as String,
      role: data['role'] as String,
    );
  }
}

File: medtech/src/main/java/com/company/medtech/auth/dto/AuthResponse.java
package com.company.medtech.auth.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class AuthResponse {

    private String token;
    private String role;
    private String id;
    private String email;
    private String fullName;
    private String avatarUrl;
}
